server/apps/microservices/src/microservices.module.ts

```diff
@@ -0,0 +1,69 @@
+import { BullModule } from '@nestjs/bull';
+import { Module } from '@nestjs/common';
+import { TypeOrmModule } from '@nestjs/typeorm';
+import { DatabaseModule } from '@app/database';
+import { AssetEntity } from '@app/database/entities/asset.entity';
+import { ExifEntity } from '@app/database/entities/exif.entity';
+import { SmartInfoEntity } from '@app/database/entities/smart-info.entity';
+import { UserEntity } from '@app/database/entities/user.entity';
+import { MicroservicesService } from './microservices.service';
+import { AssetUploadedProcessor } from './processors/asset-uploaded.processor';
+import { ThumbnailGeneratorProcessor } from './processors/thumbnail.processor';
+import { MetadataExtractionProcessor } from './processors/metadata-extraction.processor';
+import { VideoTranscodeProcessor } from './processors/video-transcode.processor';
+
+@Module({
+  imports: [
+    DatabaseModule,
+    TypeOrmModule.forFeature([UserEntity, ExifEntity, AssetEntity, SmartInfoEntity]),
+    BullModule.forRootAsync({
+      useFactory: async () => ({
+        redis: {
+          host: process.env.REDIS_HOSTNAME || 'immich_redis',
+          port: 6379,
+        },
+      }),
+    }),
+    BullModule.registerQueue({
+      name: 'thumbnail-generator-queue',
```

Comment:
There's two definitions for this queue, is that intended?

Reply:
There is no duplication on the same module. But you have to define the Queue on the module that uses it, i.e `schedule-tasks.module.ts`, otherwise you get this error 

```bash
immich-server_1            | [Nest] 158  - 06/09/2022, 1:18:43 AM   ERROR [ExceptionHandler] Nest can't resolve dependencies of the ScheduleTasksService (AssetEntityRepository, ?, BullQueue_video-conversion-queue). Please make sure that the argument BullQueue_thumbnail-generator-queue at index [1] is available in the ScheduleTasksModule context.
immich-server_1            | 
immich-server_1            | Potential solutions:
immich-server_1            | - If BullQueue_thumbnail-generator-queue is a provider, is it part of the current ScheduleTasksModule?
immich-server_1            | - If BullQueue_thumbnail-generator-queue is exported from a separate @Module, is that module imported within ScheduleTasksModule?
immich-server_1            |   @Module({
immich-server_1            |     imports: [ /* the Module containing BullQueue_thumbnail-generator-queue */ ]
immich-server_1            |   })
immich-server_1            | 
immich-server_1            | Error: Nest can't resolve dependencies of the ScheduleTasksService (AssetEntityRepository, ?, BullQueue_video-conversion-queue). Please make sure that the argument BullQueue_thumbnail-generator-queue at index [1] is available in the ScheduleTasksModule context.
immich-server_1            | 
immich-server_1            | Potential solutions:
immich-server_1            | - If BullQueue_thumbnail-generator-queue is a provider, is it part of the current ScheduleTasksModule?
immich-server_1            | - If BullQueue_thumbnail-generator-queue is exported from a separate @Module, is that module imported within ScheduleTasksModule?
immich-server_1            |   @Module({
immich-server_1            |     imports: [ /* the Module containing BullQueue_thumbnail-generator-queue */ ]
immich-server_1            |   })
immich-server_1            | 
immich-server_1            |     at Injector.lookupComponentInParentModules (/usr/src/app/node_modules/@nestjs/core/injector/injector.js:202:19)
immich-server_1            |     at processTicksAndRejections (node:internal/process/task_queues:96:5)
immich-server_1            |     at Injector.resolveComponentInstance (/usr/src/app/node_modules/@nestjs/core/injector/injector.js:157:33)
immich-server_1            |     at resolveParam (/usr/src/app/node_modules/@nestjs/core/injector/injector.js:108:38)
immich-server_1            |     at async Promise.all (index 1)
immich-server_1            |     at Injector.resolveConstructorParams (/usr/src/app/node_modules/@nestjs/core/injector/injector.js:123:27)
immich-server_1            |     at Injector.loadInstance (/usr/src/app/node_modules/@nestjs/core/injector/injector.js:52:9)
immich-server_1            |     at Injector.loadProvider (/usr/src/app/node_modules/@nestjs/core/injector/injector.js:74:9)
immich-server_1            |     at async Promise.all (index 3)
immich-server_1            |     at InstanceLoader.createInstancesOfProviders (/usr/src/app/node_modules/@nestjs/core/injector/instance-loader.js:44:9)
```

Reply:
Actually I lied... From the `microservicess.module.ts`, you can export `BullModule`. Then in `schedule-tasks.module.ts`, including the `MicroservicesModule`, then you don't have to register the queue again.

Reply:
Fair enough, it seems weird to me that you have to redefine it along with the config too, as which config does it decide to take?

Reply:
This is the part I am unsure of, so I redefine the queue to have the same setting just in case.